js/views/playingScreen.js

```diff
@@ -115,16 +109,94 @@ var PlayingScreen = React.createClass({
       this.showControlBar(event);
       this.props.controller.startHideControlBarTimer();
     }
-    else {
-      this.props.controller.togglePlayPause();
+    else if (!this.props.controller.videoVr) {
+      this.props.controller.togglePlayPause(event);
     }
   },
 
-  handlePlayerMouseMove: function() {
+  handlePlayerMouseDown: function(e) {
+    if (!this.props.controller.videoVr) {
+      return;
+    }
+    
+    this.setState({
+      isMouseDown: true,
+      XMouseStart: e.pageX,
+      YMouseStart: e.pageY
+    });
+    if (this.props.controller.onTouched) {
+      this.props.controller.onTouched();
+    }
+  },
+
+  handlePlayerMouseMove: function(e) {
     if(!this.isMobile && this.props.fullscreen) {
       this.showControlBar();
       this.props.controller.startHideControlBarTimer();
     }
+
+    if (this.props.controller.videoVr && this.state.isMouseDown) {
+      this.setState({
+        isMouseMove: true
+      });
+      
+      var params = this.getDirectionParams(e.pageX, e.pageY);
+      if (this.props.controller.onTouching) {
+        this.props.controller.onTouching(params);
+      }
+    }
+  },
+
+  handlePlayerMouseUp: function(e) {
+    // pause or play the video if the skin is clicked on desktop
+    if (!this.isMobile) {
+      e.stopPropagation(); // W3C
+      e.cancelBubble = true; // IE
+
+      this.props.controller.state.accessibilityControlsEnabled = true;
+      if (!this.props.controller.videoVr) {
+        this.props.controller.togglePlayPause();
+      }
+    }
+    // for mobile, touch is handled in handleTouchEnd
+    if (this.props.controller.videoVr) {
+      this.setState({
+        isMouseDown: false,
+      });
+      if (this.props.controller.onTouched) {
+        this.props.controller.onTouched();
+      }
+    }
+  },
+  
+  handlePlayerMouseLeave: function () {
+    if (this.props.controller.videoVr) {
+      this.setState({
+        isMouseDown: false,
+      });
+    }
+  },
+  
+  handlePlayerClicked: function (event) {
+    if(!this.state.isMouseMove){
+      this.props.controller.togglePlayPause(event);
+    }
+    
+    this.setState({
+      isMouseMove: false,
+    });
+  },
+  
+  getDirectionParams: function(pageX, pageY) {
+    var dx = pageX - this.state.XMouseStart;
+    var dy = pageY - this.state.YMouseStart;
+    var maxDegreesX = 90;
+    var maxDegreesY = 120;
+    var degreesForPixelYaw = maxDegreesX / this.props.componentWidth;
+    var degreesForPixelPitch = maxDegreesY / this.props.componentHeight;
+    var yaw = (this.props.controller.state.viewingDirection.yaw || 0) + dx * degreesForPixelYaw;
```

Comment:
Do we need validations here, or are these always assumed to be numbers? Would recommend propTypes at least

Reply:
\+